frontend/src/components/RequestSourceInputs.vue

```diff
@@ -0,0 +1,263 @@
+}
+
+interface Emits {
+  (e: 'update:modelValue', value: RequestFormResult): void;
+}
+
+const props = defineProps<Props>();
+const emits = defineEmits<Emits>();
+
+const configuration = useConfiguration();
+const ethereumProvider = useEthereumProvider();
+
+const { provider, signer } = storeToRefs(ethereumProvider);
+const { chains } = storeToRefs(configuration);
+
+const selectedAmount = ref('');
+
+const _selectedSourceChain = ref<SelectorOption<Chain> | null>(
+  getChainSelectorOption(String(provider.value?.chainId.value), chains.value),
+);
+const selectedSourceChain = computed({
+  get() {
+    return (
+      _selectedSourceChain.value ??
+      getChainSelectorOption(String(provider.value?.chainId.value), chains.value)
+    );
+  },
+  set(chain: SelectorOption<Chain> | null) {
+    _selectedSourceChain.value = chain;
+  },
+});
+watch(provider, () => updateSourceChain());
+
+const { sourceChains, switchChain } = useChainSelection(provider, chains, selectedSourceChain);
+
+const selectedSourceChainIdentifier = computed(
+  () => selectedSourceChain.value?.value.identifier ?? -1,
+);
+
+const { selectedToken, selectedTokenAddress, tokens, addTokenToProvider, addTokenAvailable } =
+  useTokenSelection(chains, selectedSourceChainIdentifier, provider);
+
+const {
+  available: showRequestFee,
+  formattedAmount: formattedRequestFeeAmount,
+  amount: requestFeeAmount,
+} = useRequestFee(
+  computed(() => selectedSourceChain.value?.value.rpcUrl),
+  computed(() => selectedSourceChain.value?.value.requestManagerAddress),
+  computed(() =>
+    selectedAmount.value && selectedToken.value
+      ? TokenAmount.parse(selectedAmount.value, selectedToken.value.value)
+      : undefined,
+  ),
+);
+
+const { available: showTokenBalance, formattedBalance: formattedTokenBalance } = useTokenBalance(
+  provider,
+  signer,
+  selectedTokenAddress,
+);
+
+const {
+  available: faucetAvailable,
+  active: faucetRequestActive,
+  run: runFaucetRequest,
+} = useFaucet(
+  signer,
+  computed(() => selectedSourceChain.value?.value.identifier),
+);
+
+const totalRequestAmount = computed(() => {
+  if (!props.modelValue.token || !requestFeeAmount.value) {
+    return '';
+  }
+  const sourceAmount = TokenAmount.parse(props.modelValue.amount, props.modelValue.token.value);
+  const total = requestFeeAmount.value?.uint256.add(sourceAmount.uint256);
+  const totalAmount = new TokenAmount({
+    amount: total.asString,
+    token: props.modelValue.token.value,
+  });
+  return totalAmount.format();
+});
+
+const updateAmount = () =>
+  emits(
+    'update:modelValue',
+    Object.assign({}, props.modelValue, { amount: selectedAmount.value }),
+  );
+const updateSourceChain = () =>
+  emits(
+    'update:modelValue',
+    Object.assign({}, props.modelValue, { sourceChain: selectedSourceChain.value }),
+  );
+const updateToken = () =>
+  emits('update:modelValue', Object.assign({}, props.modelValue, { token: selectedToken.value }));
```

Comment:
Again, somehow related to earlier comments, but this seem super complicated. Why do we need an update function for every single input. Just have some references for each input, a computed value with `get` and `set` to represent the all together as a `v-model` and than emit an event on `watch`ing this computed value. This should heavily reduce the code. From my perspective this component is actually quite simple. Or at least much simpler than what the code suggests. Possibly somehow related to that you try to use the same model for source and target chain inputs.

PS: All the `watch`ers below could be removed then too.